feat: workflow to test environment variables

patheard opened the pull request:
# Summary
Add a GitHub workflow that checks if the Example, Staging and
Production env variable names are in sync.  This is to catch cases
where environment variable changes are not properly made to
all environments.

# Related
* cds-snc/notification-planning#421
* cds-snc/notification-planning-core#10

# What are you changing?
- [ ] Releasing a new version of Notify
- [ ] Changing kubernetes configuration
- [x] GitHub workflow

mohamed-cds reviewed (changes requested):
This is really nice! One suggestion is to check that every entry in the file contains a `=`. This will prevent accidentally dumping secrets in the event of a malformed entry in a `.env` that doesn't contain a `=`.

Something like `cat test.txt| grep -v '=' | wc -l` > 0 ; abort with malformed and don't print diff

patheard commented:
That's a really good idea.  I've added in `grep="="` so that only environment variables with an equal sign will be split and compared.

I've also updated the action output so that only the `diff` is printed to further limit the chance of any accidental secret leaking.

jimleroyer reviewed (approved):
👏
